technical-reports/format/groups.md

```diff
@@ -34,251 +57,637 @@ A file MAY contain many tokens and they MAY be nested arbitrarily in groups like
+## Group Properties
 
-The names of items in a group are case sensitive. As per the guidance in the [design token chapter](#name-and-value), tools MAY display a warning to users when groups contain items whose names differ only in case and could therefore lead to naming clashes when exported.
+Groups MAY include the following properties:
 
-<p class="ednote" title="Naming practices">
-  The format editors acknowledge existing best-practices for token naming, but place no direct constraints on naming via the specification.
-</p>
+### Required Properties
 
-## Additional group properties
+None. Groups have no required properties.
 
-<div class="ednote" title="Group properties vs. nested group and token names">
+### Optional Properties
 
-To prevent collisions with token names, token properties are prefixed with a dollar sign (`$`). Using this prefix eliminates the need for a reserved words list and helps future-proof the spec.
+#### `$description`
 
-Group keys without a dollar sign (`$`) prefix denote:
+Groups MAY include an optional `$description` property, whose value MUST be a plain JSON string describing the group's purpose.
 
-- **A token name:** distinguishable by containing a `$value` property
+<aside class="example">
 
-  ```json
-  {
-    "Group of tokens": {
-      "$description": "This is an example of a group containing a single token",
-      "Token name": {
+```json
+{
+  "spacing": {
+    "$description": "All spacing-related design tokens organized by usage context",
+    "margin": {
+      /* tokens */
+    },
+    "padding": {
+      /* tokens */
+    }
+  }
+}
+```
+
+</aside>
+
+#### `$type`
+
+Groups MAY include an optional [`$type`](design-token#type) property that acts as a default type for any tokens within the group that do not explicitly declare their own type. This type inheritance applies to nested groups and their tokens unless overridden.
+
+<aside class="example">
+
+```json
+{
+  "color": {
+    "$type": "color",
+    "primary": {
+      "$value": {
+        "colorSpace": "srgb",
+        "components": [0, 0.4, 0.8],
+        "hex": "#0066cc"
+      }
+    },
+    "secondary": {
+      "$value": {
+        "colorSpace": "srgb",
+        "components": [0.4, 0.6, 1],
+        "hex": "#6699ff"
+      }
+    },
+    "semantic": {
+      "success": {
         "$value": {
           "colorSpace": "srgb",
-          "components": [0, 0, 0]
+          "components": [0, 0.8, 0.4],
+          "hex": "#00cc66"
         }
+      },
+      "warning": { "$type": "string", "$value": "amber" }
+    }
+  }
+}
+```
+
+</aside>
+
+#### `$extends`
+
+Groups MAY include an optional `$extends` property to inherit tokens and properties from another group. The `$extends` property is syntactic sugar for JSON Schema's `$ref` keyword and follows the same semantic behavior as defined in [[json-schema-2020-12]].
```

Comment:
Also, unrelated, but just for clarity specify that it‘s an error if it points to a token. This is strongly suggested by the previous sentence, but just repeating the inverse makes it clear.

```suggestion
Groups MAY include an optional `$extends` property to inherit tokens and properties from another group. `$extends` MUST NOT reference a token. The `$extends` property is syntactic sugar for JSON Schema's `$ref` keyword and follows the same semantic behavior as defined in [[json-schema-2020-12]].
```

Reply:
Good idea! Clarity is always good.